modules/stanford_earth_capx/src/EventSubscriber/EarthCapxEventsSubscriber.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Drupal\stanford_earth_capx\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\Event\MigrateEvents;
+use Drupal\migrate\Event\MigratePreRowSaveEvent;
+use Drupal\migrate\Event\MigratePostRowSaveEvent;
+use Drupal\migrate\Event\MigrateRowDeleteEvent;
+use Drupal\migrate\Row;
+use Drupal\stanford_earth_capx\EarthCapxInfo;
+
+/**
+ * Class EntityTypeSubscriber.
+ *
+* @package Drupal\stanford_earth_capx\EventSubscriber
+*/
+class EarthCapxEventsSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   *
+   * @return array
+   *   The event names to listen for, and the methods that should be executed.
+   */
+  public static function getSubscribedEvents() {
+    return [
+      MigrateEvents::PRE_ROW_SAVE => 'migratePreRowSave',
+      MigrateEvents::POST_ROW_SAVE => 'migratePostRowSave',
+      MigrateEvents::POST_ROW_DELETE => 'migratePostRowDelete',
+    ];
+  }
+
+  /**
+   * React to a migrate PRE_ROW_SAVE event.
+   * and decide if we really need to re-import a profile
+   *
+   * @param \Drupal\migrate\Event\MigratePreRowSaveEvent $event
+   *
+   */
+  public function migratePreRowSave(MigratePreRowSaveEvent $event) {
+
+    // get the row in question
+    $row = $event->getRow();
+    // see if we already have migration information for this profile
+    $info = new EarthCapxInfo($row->getSourceProperty('sunetid'));
+    // only proceed if the profile has changed (by checking etag & photo timestamp
```

Comment:
Where does the etag check happen? It's not clear in this function.

Reply:
Comment rewritten for clarity.